### PR TITLE
QoS configuration tutorial

### PR DESCRIPTION
**Pull request by Acuadros95**
Signed-off-by: acuadros95 <acuadros1995@gmail.com>

### CODE REVIEW

File: docs/rst/microros_documentation/user_api/user_api_qos.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+.. code-block:: c
+
+  // Create a publisher with depth 100.
+  rcl_publisher_t custom_publisher;
+  rmw_qos_profile_t custom_qos = rmw_qos_profile_default;
+  custom_qos.depth = 100;
+  rclc_publisher_init(
+      &custom_publisher, &node,
+      &type_support, &topic_name, &rmw_qos_profile_default);
```
Comment (pablogs9) [inline docs/rst/microros_documentation/user_api/user_api_qos.rst:28]:
```suggestion
      &type_support, &topic_name, &custom_qos);
```

File: docs/rst/microros_documentation/user_api/user_api_qos.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    - Default Publisher/subscriber: `rmw_qos_profile_default <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles.h#L51>`__.
+    - Best effort Publisher/subscriber : `rmw_qos_profile_sensor_data <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles.h#L25>`__.
+    - Default Service/client: `rmw_qos_profile_services_default <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles./h#L64>`__.
+    - Best effort Service/client: `rmw_qos_profile_services_default <https://github.com/ros2/rmw/blob/humble/rmw/include/rmw/qos_profiles.h#L64>`__ with ``reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT``
+
+Behaviour
+^^^^^^^^^
+
+When micro-ROS history slots are complete and new data arrives, the behaviour is determined by the configured ``history`` kind and ``depth``.
```
Comment (pablogs9) [inline docs/rst/microros_documentation/user_api/user_api_qos.rst:44]: Make a reference here to the place where we explain the RMW history slots

**Review by pablogs9 — CHANGES_REQUESTED** [review]
Minor changes

File: docs/rst/microros_documentation/user_api/user_api_qos.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  History kind types not listed on this section are treated as ``RMW_QOS_POLICY_HISTORY_KEEP_LAST``.
+
+Lifespan
+--------
+
+To free history slots, micro-ROS will check the lifespan of the stored data against the configured qos value.
+When the Lifespan period is reached, the history slot will be freed to be used on incoming data.
+
+.. TODO(acuadros95): Published data is handle by Fast-DDS side?
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_qos.rst:66]: @pablogs9 Thoughts about this?

File: docs/rst/microros_documentation/user_api/user_api_qos.rst (hunk 2 above)
Comment (Acuadros95) [reply docs/rst/microros_documentation/user_api/user_api_qos.rst:44]: https://github.com/eProsima/vulcanexus/pull/114/commits/1c99fa6141e145405199590760f3c010f12383cd

File: docs/rst/microros_documentation/user_api/user_api_qos.rst (hunk 1 above)
Comment (Acuadros95) [reply docs/rst/microros_documentation/user_api/user_api_qos.rst]: https://github.com/eProsima/vulcanexus/pull/114/commits/1c99fa6141e145405199590760f3c010f12383cd

**Review by dcoello-eProsima — APPROVED** [review]

**Pull request merged**